src/loki/reach.cc

```diff
@@ -53,6 +53,12 @@ directed_reach Reach::operator()(const DirectedEdge* edge,
     return reach;
   max_reach_ = max_reach;
 
+  // these are used below to get conservative estimates of forward and reverse reach
+  constexpr uint16_t forward_disallow_mask =
+      sif::kDisallowEndRestriction | sif::kDisallowSimpleRestriction | sif::kDisallowClosure;
+  constexpr uint16_t reverse_disallow_mask =
+      sif::kDisallowStartRestriction | sif::kDisallowSimpleRestriction | sif::kDisallowClosure;
```

Comment:
when doing the reach check we tell the `Allowed` (filter) method what type of conservative access we want. We used to do this manually but now we let the method do it for us